pkg/kubelet/network/dns/dns_windows.go

```diff
@@ -63,11 +64,18 @@ var (
 	procGetNetworkParams = iphlpapidll.MustFindProc("GetNetworkParams")
 )
 
+func fileExists(filename string) bool {
+	stat, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return stat.Mode().IsRegular()
+}
+
 func getHostDNSConfig(resolverConfig string) (*runtimeapi.DNSConfig, error) {
-	if resolverConfig != "" && resolverConfig != hostResolvConf {
-		err := fmt.Errorf(`Unexpected resolver config value: "%s". Expected "" or "%s".`, resolverConfig, hostResolvConf)
-		klog.ErrorS(err, "Cannot get host DNS Configuration.")
-		return nil, err
+	if fileExists(resolverConfig) {
```

Comment:
I think, for clarity, this should check `!= "" && fileExists`

Reply:
Or maybe it should check `== "" || fileExists` - this is the back-compatible path, which lets `getDNSConfig` return it's defaults.

Maybe restructure this as:

```
if resolverConfig != hostResolvConf || fileExists(resolverConfig) {
    // Get the DNS config from a resolv.conf-like file.  This handles "" by returning defaults.
    return getDNSConfig(resolverConfig)
}

// If we get here, the resolverConfig == hostResolvConf and that is not actually a file, so
// it means to use the host settings.

<...>
```

Reply:
Done